presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedInPredicateSubqueryToSemiJoin.java

```diff
@@ -106,7 +107,8 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<Void> context)
                     semiJoinSymbol,
                     Optional.empty(),
                     Optional.empty(),
-                    Optional.empty()
+                    Optional.empty(),
+                    SystemSessionProperties.isLegacySemiJoin(context.get())
```

Comment:
I think property: "if null is distinguishable from false" is more algebraic feature of `SemiJoin` rather than optimization turned on/off by feature toggle. The property can be obtained in various ways (e.g: using analysis) and other optimizers (e.g: predicate pushdown) have to take it into account.

I think you can treat this property like "partitioning" or `IndexJoin` vs `InnerJoin`. `SemiJoin` producing "ternary logic" is different node than semi join producing binary logic